draft-ietf-quic-transport.md

```diff
@@ -3838,143 +3838,135 @@ The details of loss detection and congestion control are described in
+perform ECN validation using the reported ECT(1) counts.
 
 
-#### Receiving ACK Frames {#ecn-ack}
+#### Receiving ACK Frames with ECN Counts {#ecn-ack}
 
-Erroneous application of ECN marks in the network can result in degraded
-connection performance.  An endpoint that receives an ACK frame with ECN
-counts therefore validates the counts before using them. It performs this
-validation by comparing newly received counts against those from the last
-successfully processed ACK frame. Any increase in ECN counts is validated
-based on the markings that were applied to packets that are newly
-acknowledged in the ACK frame.
+Erroneous application of CE-marks by the network can result in degraded
+connection performance.  An endpoint that receives an ACK frame with ECN counts
+therefore validates the counts before using them. It performs this validation by
+comparing newly received counts against those from the last successfully
+processed ACK frame. Any increase in the ECN counts is validated based on the
+ECN markings that were applied to packets that are newly acknowledged in the ACK
+frame.
 
-If an ACK frame newly acknowledges a packet that the endpoint sent with either
-ECT(0) or ECT(1) codepoints set, ECN validation fails if ECN counts are not
-present in the ACK frame.  This check detects a network element that zeroes out
-ECN bits or a peer that is unable to access ECN markings.
+If an ACK frame newly acknowledges a packet that the endpoint sent setting
+either the ECT(0) or ECT(1) codepoint, ECN validation fails if the ECN counts
+are not present in the ACK frame. This check detects a network element that
+zeroes the ECN field or a peer that is unable to access ECN the markings.
 
-ECN validation fails if the sum of the increase in ECT(0) and ECN-CE counts is
-less than the number of newly acknowledged packets that were originally sent
+ECN validation also fails if the sum of the increase in ECT(0) and ECN-CE counts
+is less than the number of newly acknowledged packets that were originally sent
 with an ECT(0) marking.  Similarly, ECN validation fails if the sum of the
 increases to ECT(1) and ECN-CE counts is less than the number of newly
 acknowledged packets sent with an ECT(1) marking.  These checks can detect
-removal of ECN markings in the network.
+remarking of ECN CE-markings by the network.
 
 An endpoint could miss acknowledgements for a packet when ACK frames are lost.
-It is therefore possible for the total increase in ECT(0), ECT(1), and ECN-CE
-counts to be greater than the number of packets acknowledged in an ACK frame.
-This is why counts are permitted to be larger than might be accounted for by
-newly acknowledged packets.
+It is therefore possible for the total increase in the ECT(0), ECT(1), and
+ECN-CE counts to be greater than the number of packets acknowledged in an ACK
+frame.  This is why ECN counts are permitted to be larger than the value
+corresponding to the largest acknowledged packet number.
```

Comment:
The old text makes sense to me, but I don't understand this proposed text. What value corresponding to the largest acknowledged PN is this text talking about?

Reply:
(I couldn't clearly understand the old text in the old context... but expect the intention was fine.) Could someone propose a new sentence following the structure of the old one?

Reply:
I think the original text was this - I wouldn't object to using these words, but I think they could be clearer: "This is why counts are permitted to be larger than might be accounted for by newly acknowledged packets."

Reply:
Didn't we also discuss a check were all packets get marked as CE (including non-ECT packets)? That's actually the only check that is basically implemented today in TCP (checking if the SYN is CE which should be non-ECT).

Reply:
@mirjak - Can you suggest what text you think is missing, so we can complete the edit?

Reply:
This might be more than an edit but I guess there could be another case, that checks e.g. after one RTT if all packets got CE marked and then fails validation.